frontend/src/components/request/request.js

```diff
@@ -0,0 +1,99 @@
+import React, { useState } from 'react';
+import styled from 'styled-components';
+import Button from '../../components/common/Button/Button';
+import Checkbox from './Checkbox/Checkbox';
+import PropTypes from 'prop-types';
+const RequestBlock = styled.div``;
+
+Request.propTypes = {
+  user: PropTypes.object,
+  fromList: PropTypes.array,
+  toList: PropTypes.array,
+  minimumPassenger: PropTypes.array,
+  onClickRequest: PropTypes.func,
+};
+
+function Request({ user, fromList, toList, minimumPassenger, onClickRequest }) {
+  const [from, setFrom] = useState(null);
+  const [to, setTo] = useState(null);
+  const [minPassenger, setMinPassenger] = useState(null);
+
+  const onCheckboxClickHandler = (name, value) => () => {
+    if (name === 'from') {
+      from === value ? setFrom(null) : setFrom(value);
+    } else if (name === 'to') {
+      to === value ? setTo(null) : setTo(value);
+    } else if (name === 'minimumPassenger') {
+      minPassenger === value ? setMinPassenger(null) : setMinPassenger(value);
+    }
+  };
+
+  const onButtonClickHandler = () => {
+    if (
+      user !== null &&
+      from !== null &&
+      to !== null &&
+      minPassenger !== null
+    ) {
+      onClickRequest({
+        userId: user.id,
+        from: from,
+        to: to,
+        minimumPassenger: minPassenger,
+      });
+    } else {
+      window.alert('Please Check!');
+    }
+  };
+
+  const MakeCheckbox = (name, from, to, minPassenger) =>
+    function(value) {
+      return (
+        <Checkbox
+          key={value}
+          name={name}
+          value={value}
+          onClick={onCheckboxClickHandler(name, value)}
+          checked={
+            (name === 'from' && from === value) ||
+            (name === 'to' && to === value) ||
+            (name === 'minimumPassenger' && minPassenger === value)
+          }
+        />
+      );
+    };
```

Comment:
I think `MakeCheckBox` function has too much responsibility.

I suggest removing this function and separate responsibility:

```javascript
{fromList && fromList.map(MakeCheckbox('from', from, to, minPassenger))}
```

into

```javascript
{fromList && fromList.map(value => (
  <Checkbox
          key={value}
          name={'from'}
          value={value}
          onClick={onCheckboxClickHandler(name, value)}
          checked={value !== from}
        />
))}
```

Reply:
If we change like this I think we don't need `onCheckboxClickHandler` function.

So code would be like this

from 

```javascript
onClick={onCheckboxClickHandler(name, value)}
```

into

```javascript
onClick={() => setFrom(value)}
```

something like this

Reply:
if I change the onClick function like that, We couldn't cancel the checkbox. Is it okay?